homeassistant/components/home_plus_control/__init__.py

```diff
@@ -101,12 +101,29 @@ async def async_update_data():
             # Note: asyncio.TimeoutError and aiohttp.ClientError are already
             # handled by the data update coordinator.
             async with async_timeout.timeout(10):
-                module_data = await api.async_get_modules()
+                return await api.async_get_modules()
         except HomePlusControlApiError as err:
             raise UpdateFailed(
                 f"Error communicating with API: {err} [{type(err)}]"
             ) from err
 
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="home_plus_control_module",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=300),
+    )
+    hass_entry_data[DATA_COORDINATOR] = coordinator
+
+    @callback
+    def _async_update_entities():
```

Comment:
This integration was signally to add/remove entities which rely on coordinator data before the coordinator update has finished so I moved it into a listener instead to ensure it never runs before the update is finished